streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/buffergeometry/BufferGeomProcessor.java

```diff
@@ -0,0 +1,224 @@
+
+  @Override
+  public DataProcessorDescription declareModel() {
+    return ProcessingElementBuilder.create("org.apache.streampipes.processors.geo.jvm.jts.processor.buffergeometry")
+        .category(DataProcessorType.GEO)
+        .withAssets(Assets.DOCUMENTATION, Assets.ICON)
+        .withLocales(Locales.EN)
+        .requiredStream(StreamRequirementsBuilder
+            .create()
+            .requiredPropertyWithUnaryMapping(
+                EpRequirements.domainPropertyReq("http://www.opengis.net/ont/geosparql#Geometry"),
+                Labels.withId(GEOM_KEY),
+                PropertyScope.MEASUREMENT_PROPERTY)
+            .requiredPropertyWithUnaryMapping(
+                EpRequirements.domainPropertyReq("http://data.ign.fr/def/ignf#CartesianCS"),
+                Labels.withId(EPSG_KEY),
+                PropertyScope.MEASUREMENT_PROPERTY)
+            .build())
+        .outputStrategy(OutputStrategies.append(
+                EpProperties.stringEp(
+                    Labels.withId(GEOM_KEY),
+                    GEOM_RUNTIME,
+                    "http://www.opengis.net/ont/geosparql#Geometry"
+                ),
+                EpProperties.numberEp(
+                    Labels.withId(EPSG_KEY),
+                    EPSG_RUNTIME,
+                    "http://data.ign.fr/def/ignf#CartesianCS"
+                )
+            )
+        )
+        .requiredSingleValueSelection(
+            Labels.withId(CAP_KEY),
+            Options.from(
+                CapStyle.Square.name(),
+                CapStyle.Flat.name(),
+                CapStyle.Round.name())
+        )
+        .requiredSingleValueSelection(
+            Labels.withId(JOIN_KEY),
+            Options.from(
+                JoinStyle.Bevel.name(),
+                JoinStyle.Mitre.name(),
+                JoinStyle.Round.name())
+        )
+        .requiredSingleValueSelection(
+            Labels.withId(SIDE_KEY),
+            Options.from(
+                BufferSide.Both.name(),
+                BufferSide.Left.name(),
+                BufferSide.Right.name())
+        )
+        .requiredIntegerParameter(
+            Labels.withId(MITRE_LIMIT_KEY),
+            5)
+        .requiredIntegerParameter(
+            Labels.withId(SEGMENTS_KEY),
+            8
+        )
+        .requiredFloatParameter(
+            Labels.withId(SIMPLIFY_FACTOR_KEY),
+            0.01f
+        )
+        .requiredFloatParameter(
+            Labels.withId(DISTANCE_KEY)
+        )
+        .build();
+  }
+
+  @Override
+  public void onInvocation(ProcessorParams parameters, SpOutputCollector spOutputCollector,
+                           EventProcessorRuntimeContext runtimeContext) throws SpRuntimeException {
+
+    try {
+      if (SpReprojectionBuilder.isSisConfigurationValid()){
+        LOG.info("SIS DB Settings successful checked ");
+      } else {
+        LOG.warn("The required EPSG database is not imported");
```

Comment:
Maybe you can provide some information on what to do in such a case?

Reply:
This is the same behavior as already in the Reprojection processor. This processor requires the EPSG Setup.
That is why there is an E in the symbol. The user documentation is still missing on the website, so the URL cannot be referenced here. So this is a future to do.

Reply:
A lot of future todos, I trust you to keep them all on your radar :)